src/Imagine/Cache/Resolver/ProxyResolver.php

```diff
@@ -56,7 +56,7 @@ public function remove(array $paths, array $filters): void
         $this->resolver->remove($paths, $filters);
     }
 
-    protected function rewriteUrl(string $url): string
+    protected function rewriteUrl(string $url): ?string
```

Comment:
the only way how this can return null is when there is a regex error in the `preg_replace` call. is the calling side of this handling that? if not, i'd prefer to check the return value of preg_replace in this method and use preg_last_error_msg to throw an exception. then we can leave this as returning a non-nullable string.